src/main/java/dev/waterdog/waterdogpe/network/downstream/SwitchDownstreamHandler.java

```diff
@@ -129,6 +158,10 @@ public final boolean handle(StartGamePacket packet) {
         }
         entities.clear();
 
+        ObjectSet<ScoreInfo> scoreInfos = this.player.getScoreInfos();
+        injectRemoveScoreInfos(this.player.getUpstream(), scoreInfos);
+        scoreInfos.clear();
+
```

Comment:
Required to remove old score infos from the player (could check if score infos is not empy? idk)